### PR TITLE
[invoke] 🐛 check keypair only when required in config

### PR DESCRIPTION
**Pull request by pducolin**
What does this PR do?
---------------------

Go through keypair validation only when required in config

Which scenarios this will impact?
-------------------

Motivation
----------

Additional Notes
----------------

### CODE REVIEW

File: tasks/vm.py [hunk 1]
```diff
@@ -39,7 +39,6 @@ def create_vm(
     deploy(
         ctx,
         scenario_name,
-        key_pair_required=True,
```
Comment (ogaca-dd) [inline tasks/vm.py:42]: This variable is set at `True` when a scenario need a key pair so we must keep it.

File: tasks/deploy.py [hunk 2]
```diff
@@ -42,7 +41,7 @@ def deploy(
     if install_agent:
         flags["ddagent:apiKey"] = _get_api_key()
 
-    if key_pair_required:
+    if cfg.get_options().checkKeyPair:
```
Comment (ogaca-dd) [inline tasks/deploy.py:44]:
```suggestion
    if key_pair_required and cfg.get_options().checkKeyPair:
```

**Review by ogaca-dd — APPROVED** [review]
Thanks for the fix.

File: tasks/deploy.py (hunk 2 above)
Comment (pducolin) [reply tasks/deploy.py:44]: Ooops, thank you !

**Pull request merged**